src/frontend/Sider/Behandling/Vilkårvurdering/EndreDelvilkår.tsx

```diff
@@ -39,147 +32,87 @@ const DelvilkårContainer = styled.div<{ $erUndervilkår: boolean }>`
-        );
-
-        const oppdaterteSvarMedKopiertBegrunnelse = kopierBegrunnelse(
-            vurderinger,
-            oppdaterteSvarMedNesteRegel,
-            nyttSvar,
-            svaralternativer,
-            regler
-        );
-
-        oppdaterVilkårsvar(delvilkårIndex, oppdaterteSvarMedKopiertBegrunnelse);
-    };
+    useEffect(() => {
+        settBrukervurderinger(mapTilVurderingInput(delvilkårsett));
+    }, [delvilkårsett]);
 
-    const [feilmeldinger, settFeilmeldinger] = useState<Feilmeldinger>({});
+    const oppdaterteVurderinger = oppdaterVurderinger(delvilkårsett, brukervurderinger);
 
-    const validerOgLagreVilkårsvurderinger = (event: React.FormEvent<HTMLFormElement>) => {
+    const validerOgLagreVilkårsvurdering = (event: React.FormEvent<HTMLFormElement>) => {
         event.preventDefault();
 
-        const valideringsfeil = validerVilkårsvurderinger(delvilkårsett, regler);
-
+        const valideringsfeil = validerVilkårsvurdering(oppdaterteVurderinger);
         settFeilmeldinger(valideringsfeil);
 
         if (erTomtObjekt(valideringsfeil)) {
-            oppdaterVilkår({
-                id: vilkår.id,
-                behandlingId: vilkår.behandlingId,
-                delvilkårsett: delvilkårsett,
-            });
+            lagreVilkårsvurdering(oppdaterteVurderinger);
         }
     };
 
+    const oppdaterSvar = (regelId: RegelId, nyttSvar: SvarId) => {
+        settBrukervurderinger({
+            ...brukervurderinger,
+            [regelId]: { ...brukervurderinger[regelId], svar: nyttSvar || null },
+        });
+    };
+
+    const oppdaterBegrunnelse = (regelId: RegelId, nyBegrunnelse?: string) => {
+        settBrukervurderinger({
+            ...brukervurderinger,
+            [regelId]: { ...brukervurderinger[regelId], begrunnelse: nyBegrunnelse || null },
+        });
+    };
+
     const nullstillFeilmelding = (regelId: string) => {
         settFeilmeldinger({ ...feilmeldinger, [regelId]: undefined });
     };
 
     return (
-        <form onSubmit={validerOgLagreVilkårsvurderinger}>
+        <form onSubmit={validerOgLagreVilkårsvurdering}>
             <VStack gap="4">
-                {delvilkårsett.map((delvikår, delvilkårIndex) => {
-                    return delvikår.vurderinger.map((svar, indeks) => {
-                        const gjeldendeRegel = regler[svar.regelId];
-                        const erUndervilkår = indeks !== 0;
+                {delvilkårSomErRelevante(oppdaterteVurderinger).map(
+                    ([regel, delvilkår], indeks) => {
+                        const gjeldendeSvar = delvilkår.svar;
+
+                        const begrunnelsestype = gjeldendeSvar
+                            ? delvilkår.svaralternativer[gjeldendeSvar]?.begrunnelsestype
+                            : 'VALGFRI';
+
+                        const følgerFraOverordnetValg = delvilkår.følgerFraOverordnetValg !== null;
```

Comment:
Ref kommentar i backend, nå skjønner jeg hvorfor den er ønskelig.
Samtidig kan man løse det med å kun ha peker fra Spm1 -> Spm2, som jeg føler er enklere å tolke/lese

Reply:
Ja, enig i at peker fra Spm1 -> Spm2 nok ville vært mer forståelig i dette caset. Angrer litt på den i etterkant, men tenker det er litt for mye jobb nå å endre på 😅